presto-main/src/main/java/com/facebook/presto/operator/aggregation/partial/SkipAggregationBuilder.java

```diff
@@ -0,0 +1,183 @@
+import com.facebook.presto.operator.aggregation.AccumulatorFactory;
+import com.facebook.presto.operator.aggregation.builder.HashAggregationBuilder;
+import com.facebook.presto.spi.function.aggregation.GroupByIdBlock;
+import com.facebook.presto.spi.function.aggregation.GroupedAccumulator;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.common.Utils.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link HashAggregationBuilder} that does not aggregate input rows at all.
+ * It passes the input pages, augmented with initial accumulator state to the output.
+ * It can only be used at the partial aggregation step as it relies on rows be aggregated at the final step.
+ */
+public class SkipAggregationBuilder
+        implements HashAggregationBuilder
+{
+    private final LocalMemoryContext memoryContext;
+    private final List<GroupedAccumulator> groupedAccumulators;
+    @Nullable
+    private Page currentPage;
+    private final int[] hashChannels;
+
+    public SkipAggregationBuilder(
+            List<Integer> groupByChannels,
+            Optional<Integer> inputHashChannel,
+            List<AccumulatorFactory> accumulatorFactories,
+            LocalMemoryContext memoryContext)
+    {
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+        this.groupedAccumulators = accumulatorFactories.stream()
+                .map(accumulatorFactory -> accumulatorFactory.createGroupedAccumulator(UpdateMemory.NOOP))
+                .collect(toImmutableList());
+        this.hashChannels = new int[groupByChannels.size() + (inputHashChannel.isPresent() ? 1 : 0)];
+        for (int i = 0; i < groupByChannels.size(); i++) {
+            hashChannels[i] = groupByChannels.get(i);
+        }
+        inputHashChannel.ifPresent(channelIndex -> hashChannels[groupByChannels.size()] = channelIndex);
+    }
+
+    @Override
+    public Work<?> processPage(Page page)
+    {
+        checkArgument(currentPage == null);
+        currentPage = page;
+        return new CompletedWork<>();
+    }
+
+    @Override
+    public WorkProcessor<Page> buildResult()
+    {
+        if (currentPage == null) {
+            return WorkProcessor.of();
+        }
+
+        Page result = buildOutputPage(currentPage);
+        currentPage = null;
+        return WorkProcessor.of(result);
+    }
+
+    @Override
+    public boolean isFull()
+    {
+        return currentPage != null;
+    }
+
+    @Override
+    public void updateMemory()
+    {
+        if (currentPage != null) {
+            memoryContext.setBytes(currentPage.getSizeInBytes());
+        }
+    }
+
+    @Override
+    public void recordHashCollisions(HashCollisionsCounter hashCollisionsCounter)
+    {
+        // no op
```

Comment:
is this comment superfluous ?

Reply:
We can remove.

Reply:
Removed. But will note that there are tons of places where we have no op as single line comment